pinot-common/src/main/java/org/apache/pinot/sql/parsers/rewriter/AliasApplier.java

```diff
@@ -31,38 +29,30 @@
 
 
 public class AliasApplier implements QueryRewriter {
+
   @Override
   public PinotQuery rewrite(PinotQuery pinotQuery) {
-
-    // Update alias
-    Map<Identifier, Expression> aliasMap = extractAlias(pinotQuery.getSelectList());
+    Map<String, Expression> aliasMap = extractAlias(pinotQuery.getSelectList());
     applyAlias(aliasMap, pinotQuery);
-
-    // Validate
-    validateSelectionClause(aliasMap, pinotQuery);
     return pinotQuery;
   }
 
-  private static Map<Identifier, Expression> extractAlias(List<Expression> expressions) {
-    Map<Identifier, Expression> aliasMap = new HashMap<>();
-    for (Expression expression : expressions) {
-      Function functionCall = expression.getFunctionCall();
-      if (functionCall == null) {
+  private static Map<String, Expression> extractAlias(List<Expression> selectExpressions) {
+    Map<String, Expression> aliasMap = new HashMap<>();
+    for (Expression expression : selectExpressions) {
+      Function function = expression.getFunctionCall();
+      if (function == null || !function.getOperator().equals("as")) {
         continue;
       }
-      if (functionCall.getOperator().equals("as")) {
-        Expression identifierExpr = functionCall.getOperands().get(1);
-        aliasMap.put(identifierExpr.getIdentifier(), functionCall.getOperands().get(0));
+      String alias = function.getOperands().get(1).getIdentifier().getName();
+      if (aliasMap.put(alias, function.getOperands().get(0)) != null) {
+        throw new SqlCompilationException("Find duplicate alias: " + alias);
       }
     }
     return aliasMap;
   }
 
-  private static void applyAlias(Map<Identifier, Expression> aliasMap, PinotQuery pinotQuery) {
-    Expression filterExpression = pinotQuery.getFilterExpression();
```

Comment:
My concern here is that this will silently return empty results once users upgrade.

Shall we instead throw exception if detected the alias in filter and explicitly mention that alias shouldn't be used, so users can at least find out the root cause and either roll back or fix queries before upgrade?